### PR TITLE
Add error threshold and verify domain name for instance in helix tool

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This PR introduces two changes
1. Add error or recover threshold to the cluster config
2. Verify that all the instances have domain field before enabling topology awareness for cluster.

### CODE REVIEW

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java [hunk 1]
```diff
@@ -2306,6 +2326,7 @@ static class ClusterConfigFields {
     private int partitionDiskWeightInGB;
     private int maxPartitionInTransition;
     private int delayRebalanceTimeInMs;
+    private int errorOrRecoverPartitionThresholdForLoadBalance;
```
Comment (Arun-LinkedIn) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:2329]: Nit: I added a default configuration file for reference at `config/WagedRebalancerHelixConfig.json`. May be we can add this new config field there too?

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java [hunk 2]
```diff
@@ -1020,6 +1021,20 @@ private String getResourceTag(String resource) {
    * @param clusterConfigFields disk capacity of each partition.
    */
   private void setClusterConfig(ConfigAccessor configAccessor, ClusterConfigFields clusterConfigFields) {
+    // We will enable TOPOLOGY AWARE in cluster config, we need to make sure all the instances have "DOMAIN" fields
+    HelixConfigScope scope =
+        new HelixConfigScope(HelixConfigScope.ConfigScopeProperty.PARTICIPANT, Collections.singletonList(clusterName),
+            null);
+    // This will list all the participant names
+    List<String> instanceNames = configAccessor.getKeys(scope);
```
Comment (Arun-LinkedIn) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:1029]: Hmm, when `--resources` input is given to this tool, I called `setInstanceConfig` only for the instances hosting partitions of those resources (`line number 894`). But, `configAccessor.getKeys(scope)` may get all instances in the cluster. May be we need to set DOMAIN field for all instances in the cluster (irrespective of inputted resources) now ?

Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:1029]: we should probably do that, let me update it.

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:2337]: done

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:1028]: done

**Review by SophieGuo410 — APPROVED** [review]
lgtm.

**Pull request closed without merging**